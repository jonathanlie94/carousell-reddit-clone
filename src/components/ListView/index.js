import React, { Component } from 'react';
import styled from 'styled-components';
import { theme } from 'ui';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  border-radius: 8px;
  border: 1px solid ${theme.colors.grey};
  overflow: hidden;
`;

class ListView extends Component {
  render() {
    return (
      <Wrapper className={this.props.className}>
        {this.props.children}
      </Wrapper>
    );
  }
}

ListView.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

export default ListView;
