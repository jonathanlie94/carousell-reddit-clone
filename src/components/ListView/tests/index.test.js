import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import ListView from '../';

describe('<ListView />', () => {
  it('should render children when passed in', () => {
    const renderedComponent = shallow(
      <ListView>
        <div className="unique-class-name" />
      </ListView>
    );
    expect(
      renderedComponent.contains(<div className="unique-class-name" />)
    ).toEqual(true);
  });

  it('should match previous snapshot', () => {
    const renderedComponent = mount(<ListView />);
    const tree = toJSON(renderedComponent);
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
