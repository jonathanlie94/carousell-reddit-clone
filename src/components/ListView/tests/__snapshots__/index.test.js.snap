// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListView /> should match previous snapshot 1`] = `
.cqALui {
  border-radius: 8px;
  border: 1px solid #C8C8C8;
  overflow: hidden;
}

<ListView>
  <styled.div>
    <div
      className="sc-bdVaJa cqALui"
    />
  </styled.div>
</ListView>
`;
